digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadNameDAO.scala

```diff
@@ -132,21 +133,16 @@ object RoadNameDAO {
     * @param since
     * @return
     */
-  def getUpdatedRoadNames(since: DateTime): Seq[RoadName] = {
-    if (since != null) {
-      val sinceString = since.toString("yyyy-MM-dd")
-      val query =
-        s"""
+  def getUpdatedRoadNames(since: DateTime, until: Option[DateTime]): Seq[RoadName] = {
+    val untilString = if (until.nonEmpty) s"AND CREATED_TIME <= to_timestamp('${new Timestamp(until.get.getMillis)}', 'YYYY-MM-DD HH24:MI:SS.FF')" else s""
+        sql"""
         SELECT * FROM road_names
         WHERE road_number IN (
             SELECT DISTINCT road_number FROM road_names
-            WHERE valid_to IS NULL AND CREATED_TIME >= TO_DATE('${sinceString}', 'RRRR-MM-dd')
+            WHERE valid_to IS NULL AND CREATED_TIME >= $since #$untilString
```

Comment:
Why there is this hash -character?: **#**$untilString

Reply:
Because is the only way to concat a string in a sql""""""